packages/esm-appointments-app/src/header/appointments-header.component.tsx

```diff
@@ -19,6 +19,26 @@ const AppointmentsHeader: React.FC<AppointmentHeaderProps> = ({ title, appointme
   const { selectedDate, setSelectedDate } = useContext(SelectedDateContext);
   const { serviceTypes } = useAppointmentServices();
 
+  const items = [...serviceTypes];
+  const [selectedItems, setSelectedItems] = useState([]);
+
+  const handleMultiSelectChange = useCallback(
+    ({ selectedItems }) => {
+      const selectedUuids = selectedItems.map((item) => item.id);
+      if (selectedUuids.length === 0) {
+        setSelectedItems([]);
+      } else {
+        setSelectedItems(selectedUuids);
+        onChange?.(selectedUuids);
+      }
+    },
+    [onChange],
+  );
+
+  useEffect(() => {
+    onChange?.('');
+  }, [onChange]);
+
```

Comment:
```suggestion
  const serviceTypeOptions = useMemo(
    () => serviceTypes?.map((item) => ({ id: item.uuid, label: item.name })) ?? [],
    [serviceTypes],
  );
```

Let's memoize the `serviceTypes` mapping.